### DISCUSSION TITLE
preferred way of adding python package dependencies?

### DISCUSSION THREAD

**Discussion by vgeshiktor** [bentoml/BentoML#1408] [General]
what is preferred way of adding dependencies to BentoML, mostly additional python packages?

**Comment by parano** [comment]
Hi @vgeshiktor - the recommended way is letting BentoML automatically discover required python packages with the `infer_pip_packages=True` option, as demonstrated in the qucikstart guide, e.g.:

```python
@env(infer_pip_packages=True)
@artifacts([SklearnModelArtifact('my_model')])
class MyPredictionService(BentoService):

    @api(input=DataframeInput(orient="records"), batch=True)
    def predict(self, df: pd.DataFrame):
        model_input = preprocess(df)
        return self.artifacts.my_model.predict(model_input)
```
